File: app/src/main/java/org/doomer/qnotez/db/AppDatabase.java
package org.doomer.qnotez.db;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.migration.Migration;
import android.content.Context;

@Database(entities = {NoteModel.class}, version = 2)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase INSTANCE;
    public static final String DB_NAME = "qnotez.db3";

    public static AppDatabase getDatabase(Context context) {
        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(
                    context.getApplicationContext(), AppDatabase.class, DB_NAME
            ).addMigrations(AppDatabase.MIGRATION_1_2).build();
        }

        return INSTANCE;
    }

    public static void destroyInstance() {
        INSTANCE = null;
    }

    public abstract NoteDao getNoteModel();

    public static final Migration MIGRATION_1_2 = new Migration(1, 2) {

        @Override
        public void migrate(final SupportSQLiteDatabase database) {
            database.execSQL("ALTER TABLE " + NoteModel.TABLE_NAME + " ADD COLUMN " + NoteModel.COL_TRASH + " INTEGER DEFAULT 0 NOT NULL");

        }
    };
}

File: app/src/main/kotlin/org/doomer/qnotez/db/AppDatabase.kt
package org.doomer.qnotez.db

import android.arch.persistence.db.SupportSQLiteDatabase
import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.migration.Migration

val DB_NAME = "qnotez.db3"

@Database(entities = arrayOf(NoteModel::class), version = 2, exportSchema = true)
abstract class AppDatabase : RoomDatabase() {

    abstract fun noteModel() : NoteDao

    val MIGRATION_1_2: Migration = object : Migration(1, 2) {

        override fun migrate(database: SupportSQLiteDatabase) {
            database.execSQL("ALTER TABLE " + TABLE_NOTES + " ADD COLUMN " + COL_TRASH + " INTEGER DEFAULT 0 NOT NULL")

        }
    }
}

